sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/spark.py

```diff
@@ -526,7 +606,6 @@ def _validate_entry_exist(self, raise_error=False) -> MutableValidationResult:
     def _validate_fields(self) -> None:
```

Comment:
shall we just move the call for this method from `_to_rest_object` to `_customized_validate`? Even a simple try catch may help us to avoid maintaining validation logic in multiple places.

Reply:
I put it here, because we can reuse validation function added by spark teams. This validation function is also used in standalone job

Reply:
hmm... There is no limitation that we can't use shared function in `_customize_validate`.
They can be referred like below:
```python
def _customize_validate(self):
  validation_result = super()._customize_validate()
  try:
    self._validate_fields()
  except ValidationException as e:
    validation_result.append_error(yaml_path="*", message=str(e))
```

Whatever, we can do such refactor in further change.

Reply:
Thanks, will update in the next pr